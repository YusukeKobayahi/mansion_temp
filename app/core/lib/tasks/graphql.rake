# frozen_string_literal: true

module GraphqlTask
  extend Rake::DSL

  namespace :graphql do
    namespace :consumer do
      task schema_dump: :environment do
        require 'graphql/rake_task'

        GraphQL::RakeTask.new(
          load_schema: lambda { |_task|
            require File.expand_path('../../app/graphql/consumer/inno_schema', __dir__)
            Consumer::InnoSchema
          },
          directory: './graphql/consumer',
        )
        Rake::Task['graphql:schema:idl'].invoke
      end
    end
  end
end
