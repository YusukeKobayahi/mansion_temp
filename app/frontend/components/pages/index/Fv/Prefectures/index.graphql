fragment MsLibraryIndexPrefecture on Prefecture {
  id
  name
  jisCode
}
