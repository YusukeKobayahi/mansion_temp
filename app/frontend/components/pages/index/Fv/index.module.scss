@import "~/styles/variables";
@import "~/styles/utils";

.main {
  .label {
    display: none;
    @include screenMd {
      display: block;
      margin-top: 25px;
      text-align: center;
      font-weight: bold;
    }

    span {
      display: inline-block;
      line-height: 22px;
      padding: 0 7px;
      background-color: #ffffffc7;
      color: #646464;
      font-size: 14px;
    }
  }

  h1 {
    margin-top: 60px;
    text-align: center;
    font-size: 50px;
    color: #f7f7f7;
    line-height: 1.44;
    @include screenMd {
      margin-top: 15px;
      font-size: 28px;
      line-height: 1.41;
      font-weight: bold;
    }
  }
}

.searchBlock {
  @include screenMd {
    margin-top: 20px;
    background-color: #ffffff;
  }
}

.index {
  margin: 0 calc(50% - 50vw);
  padding: 0 calc(50vw - 50%);

  @include screenMd {
    padding: 0;
  }
  background-color: #ffffff;
}
