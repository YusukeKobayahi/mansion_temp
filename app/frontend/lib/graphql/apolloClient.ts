// 参考
// https://github.com/vercel/next.js/tree/canary/examples/with-apollo

import { useMemo } from "react";
import { ApolloClient, ApolloLink, InMemoryCache } from "@apollo/client";
import { createHttpLink } from "apollo-link-http";
import { from } from "apollo-link";
import { createFragmentArgumentLink } from "apollo-link-fragment-argument";
import { mergeDeep, relayStylePagination } from "@apollo/client/utilities";
import { isSSR } from "~/lib/utils";

/* eslint-disable @typescript-eslint/no-explicit-any -- キャッシュの型を事前に決められないため */
type CacheState = Record<string, any> | null;
/* eslint-enable @typescript-eslint/no-explicit-any */

/* eslint-disable @typescript-eslint/naming-convention -- GraphQLのタイプ名 (このconventionに従わない)を指定したいので */
const createCache = () =>
  new InMemoryCache({
    typePolicies: {
      Mansion: {
        keyFields: ["uniqueCode"],
        fields: {
          salesHistories: relayStylePagination(),
        },
      },
    },
  });
/* eslint-enable @typescript-eslint/naming-convention */

const link = from([
  createFragmentArgumentLink(),
  createHttpLink({
    uri: `${process.env.NEXT_PUBLIC_CORE_API_URL}/consumer/graphql`,
    fetchOptions: {
      mode: "cors",
    },
  }),
]);

function createApolloClient() {
  return new ApolloClient({
    ssrMode: isSSR(),
    // See: https://github.com/apollographql/apollo-link/issues/1258#issuecomment-604031488
    // TODO: @apollo/clientバージョン上げたら直るかも
    link: (link as unknown) as ApolloLink,
    cache: createCache(),
    defaultOptions: {
      watchQuery: {
        // SSR時にuseQueryによってリクエストしてレンダリングする場合、
        // getServerSidePropsからキャッシュが渡っている必要がある。(でないと描画されない)
        // そのため、SSR時は常にcacheから取得されるべきで、実装意図を明確にするためPolicyも設定しておく
        // (エラーになるわけではない)
        fetchPolicy: isSSR() ? "cache-only" : "cache-first",
      },
    },
  });
}

let apolloClient: ReturnType<typeof createApolloClient>;

export function initializeApollo(
  initialState: CacheState = null
): ApolloClient<CacheState> {
  const _apolloClient = apolloClient ?? createApolloClient();

  // If your page has Next.js data fetching methods that use Apollo Client, the initial state
  // gets hydrated here
  if (initialState) {
    // Get existing cache, loaded during client side data fetching
    const existingCache = _apolloClient.extract();

    // Merge the existing cache into data passed from getStaticProps/getServerSideProps
    const data = mergeDeep(initialState, existingCache);

    // Restore the cache with the merged data
    _apolloClient.cache.restore(data);
  }
  // For SSG and SSR always create a new Apollo Client
  if (isSSR()) return _apolloClient;
  // Create the Apollo Client once in the client
  if (!apolloClient) apolloClient = _apolloClient;

  return _apolloClient;
}

export function useApollo(initialState: CacheState): ApolloClient<CacheState> {
  const store = useMemo(() => initializeApollo(initialState), [initialState]);
  return store;
}
