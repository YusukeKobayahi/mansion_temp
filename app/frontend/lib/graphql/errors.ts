export interface GraphQLError {
  graphQLErrors?: {
    message: string;
    extensions?: {
      code: string;
    };
  }[];
}

export const isNotFoundError = (err: GraphQLError): boolean => {
  const graphQLErrors = err.graphQLErrors;
  if (graphQLErrors == null) return false;

  return graphQLErrors[0]?.extensions?.code === "NOT_FOUND";
};
