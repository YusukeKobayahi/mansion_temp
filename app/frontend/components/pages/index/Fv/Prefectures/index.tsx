import Link from "next/link";
import Image from "next/image";
import { MsLibraryIndexPrefectureFragment } from "~/graphql/generated";
import { buildLibraryPagePath } from "~/lib/utils";
import compact from "lodash/compact";
import { event } from "~/lib/gtag";

import styles from "./index.module.scss";

export interface PrefecturesProps {
  data: MsLibraryIndexPrefectureFragment[];
}

const Prefectures: React.FC<PrefecturesProps> = ({
  data,
}: PrefecturesProps) => {
  const sortedIds = ["13", "14", "11", "12", "27", "28", "23", "40"];

  const sortedPrefectures: MsLibraryIndexPrefectureFragment[] = compact(
    sortedIds.map((id) => data.find((pref) => pref.id === id))
  );

  const imgClassMapping: Record<string, string> = {
    "11": "saitama",
    "12": "chiba",
    "13": "tokyo",
    "14": "kanagawa",
    "23": "aichi",
    "27": "osaka",
    "28": "hyogo",
    "40": "fukuoka",
  };

  const clickPref = () => {
    event({
      action: "msl_area_click",
      category: "msl_top",
      label: "msl_pref",
      value: 1,
    });
  };

  return (
    <section className={styles.container} id="prefectures">
      <h2 className={styles.title}>エリアから探す</h2>
      <ul className={styles.list}>
        {sortedPrefectures.map((pref) => {
          const imgSrc = `/ms-library/images/thum_${
            imgClassMapping[pref.id]
          }.jpg`;
          return (
            <li key={pref.id} className={styles.item}>
              <div className={styles.image}>
                <Image
                  src={imgSrc}
                  alt={pref.name}
                  width={235}
                  height={259}
                  loading={"eager"}
                />
              </div>
              <Link
                href={"/prefectures/[jisCode]"}
                as={buildLibraryPagePath("prefectures", pref.jisCode)}
                prefetch={false}
                passHref
              >
                <button className={styles.link} onClick={clickPref}>
                  {pref.name}
                </button>
              </Link>
            </li>
          );
        })}
      </ul>
    </section>
  );
};

export default Prefectures;
