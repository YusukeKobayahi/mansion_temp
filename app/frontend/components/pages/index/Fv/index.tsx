import styles from "./index.module.scss";
import Prefectures from "~/components/pages/index/Fv/Prefectures";
import Railways from "~/components/pages/index/Railways";
import FreeWord from "~/components/pages/index/FreeWord";

import Breadcrumb, {
  BreadcrumbItemProps,
} from "~/components/commons/Breadcrumb";
import { ComponentProps } from "react";

export type FvProps = ComponentProps<typeof Railways> &
  ComponentProps<typeof FreeWord> &
  ComponentProps<typeof Prefectures> & {
    breadcrumbs: BreadcrumbItemProps[];
  };

const Fv: React.FC<FvProps> = ({ data, breadcrumbs, setTab }: FvProps) => {
  return (
    <section className={styles.container}>
      <div className={styles.main}>
        <Breadcrumb mainClassName={styles.index} breadcrumbs={breadcrumbs} />
        <p className={styles.label}>
          <span>マンション売却をご検討の方へ</span>
        </p>
        <h1>
          物件情報やノウハウが
          <br />
          手に入る
        </h1>
        <div className={styles.searchBlock}>
          <Prefectures data={data} />
          <Railways setTab={setTab} />
          <FreeWord />
        </div>
      </div>
    </section>
  );
};

export default Fv;
