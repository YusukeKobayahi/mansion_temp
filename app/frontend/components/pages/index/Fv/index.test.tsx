import { render } from "@testing-library/react";
import mockData from "~/components/pages/index/fixtures/mockData.json";
import { IndexQuery } from "~/graphql/generated";
import Fv from "~/components/pages/index/Fv";
import ToggleContextProvider from "~/components/Context/ToggleContext";
import SearchConditionContextProvider from "~/components/Context/SearchConditionContext";

describe("components/pages/index/Fv", () => {
  it("正常に表示される", () => {
    const mock = mockData as IndexQuery;
    const breadcrumbs = [
      {
        name: "Housiiトップ",
        path: "https://ieul.jp/buy/",
      },
      {
        name: "マンションライブラリー",
        path: "/ms-library/",
      },
    ];

    const { asFragment } = render(
      <SearchConditionContextProvider searchCondition={{}}>
        <ToggleContextProvider>
          <Fv
            data={mock.prefectures}
            breadcrumbs={breadcrumbs}
            setTab={() => {
              return;
            }}
          />
        </ToggleContextProvider>
      </SearchConditionContextProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
