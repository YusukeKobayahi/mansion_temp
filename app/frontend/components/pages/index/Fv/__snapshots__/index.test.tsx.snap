// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/index/Fv 正常に表示される 1`] = `
<DocumentFragment>
  <section
    class="container"
  >
    <div
      class="main"
    >
      <nav
        class="index"
      >
        <ol
          class="list"
        >
          <li
            class="item"
          >
            <a
              href="/"
            >
              イエウールトップ
            </a>
          </li>
          <li
            class="item"
          >
            <a
              href="https://ieul.jp/buy/"
            >
              Housiiトップ
            </a>
          </li>
          <li
            class="item"
          >
            <span>
              マンションライブラリー
            </span>
          </li>
        </ol>
      </nav>
      <p
        class="label"
      >
        <span>
          マンション売却をご検討の方へ
        </span>
      </p>
      <h1>
        物件情報やノウハウが
        <br />
        手に入る
      </h1>
      <div
        class="searchBlock"
      >
        <section
          class="container"
          id="prefectures"
        >
          <h2
            class="title"
          >
            エリアから探す
          </h2>
          <ul
            class="list"
          >
            <li
              class="item"
            >
              <div
                class="image"
              >
                <div
                  style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
                >
                  <div
                    style="box-sizing: border-box; display: block; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjM1IiBoZWlnaHQ9IjI1OSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                      style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                    />
                  </div>
                  <img
                    alt="東京都"
                    data-nimg="true"
                    decoding="async"
                    src="/_next/image?url=%2Fms-library%2Fimages%2Fthum_tokyo.jpg&w=640&q=75"
                    srcset="/_next/image?url=%2Fms-library%2Fimages%2Fthum_tokyo.jpg&w=256&q=75 1x, /_next/image?url=%2Fms-library%2Fimages%2Fthum_tokyo.jpg&w=640&q=75 2x"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                </div>
              </div>
              <button
                class="link"
                href="/prefectures/13000"
              >
                東京都
              </button>
            </li>
            <li
              class="item"
            >
              <div
                class="image"
              >
                <div
                  style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
                >
                  <div
                    style="box-sizing: border-box; display: block; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjM1IiBoZWlnaHQ9IjI1OSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                      style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                    />
                  </div>
                  <img
                    alt="神奈川県"
                    data-nimg="true"
                    decoding="async"
                    src="/_next/image?url=%2Fms-library%2Fimages%2Fthum_kanagawa.jpg&w=640&q=75"
                    srcset="/_next/image?url=%2Fms-library%2Fimages%2Fthum_kanagawa.jpg&w=256&q=75 1x, /_next/image?url=%2Fms-library%2Fimages%2Fthum_kanagawa.jpg&w=640&q=75 2x"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                </div>
              </div>
              <button
                class="link"
                href="/prefectures/14000"
              >
                神奈川県
              </button>
            </li>
            <li
              class="item"
            >
              <div
                class="image"
              >
                <div
                  style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
                >
                  <div
                    style="box-sizing: border-box; display: block; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjM1IiBoZWlnaHQ9IjI1OSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                      style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                    />
                  </div>
                  <img
                    alt="埼玉県"
                    data-nimg="true"
                    decoding="async"
                    src="/_next/image?url=%2Fms-library%2Fimages%2Fthum_saitama.jpg&w=640&q=75"
                    srcset="/_next/image?url=%2Fms-library%2Fimages%2Fthum_saitama.jpg&w=256&q=75 1x, /_next/image?url=%2Fms-library%2Fimages%2Fthum_saitama.jpg&w=640&q=75 2x"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                </div>
              </div>
              <button
                class="link"
                href="/prefectures/11000"
              >
                埼玉県
              </button>
            </li>
            <li
              class="item"
            >
              <div
                class="image"
              >
                <div
                  style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
                >
                  <div
                    style="box-sizing: border-box; display: block; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjM1IiBoZWlnaHQ9IjI1OSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                      style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                    />
                  </div>
                  <img
                    alt="千葉県"
                    data-nimg="true"
                    decoding="async"
                    src="/_next/image?url=%2Fms-library%2Fimages%2Fthum_chiba.jpg&w=640&q=75"
                    srcset="/_next/image?url=%2Fms-library%2Fimages%2Fthum_chiba.jpg&w=256&q=75 1x, /_next/image?url=%2Fms-library%2Fimages%2Fthum_chiba.jpg&w=640&q=75 2x"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                </div>
              </div>
              <button
                class="link"
                href="/prefectures/12000"
              >
                千葉県
              </button>
            </li>
            <li
              class="item"
            >
              <div
                class="image"
              >
                <div
                  style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
                >
                  <div
                    style="box-sizing: border-box; display: block; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjM1IiBoZWlnaHQ9IjI1OSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                      style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                    />
                  </div>
                  <img
                    alt="大阪府"
                    data-nimg="true"
                    decoding="async"
                    src="/_next/image?url=%2Fms-library%2Fimages%2Fthum_osaka.jpg&w=640&q=75"
                    srcset="/_next/image?url=%2Fms-library%2Fimages%2Fthum_osaka.jpg&w=256&q=75 1x, /_next/image?url=%2Fms-library%2Fimages%2Fthum_osaka.jpg&w=640&q=75 2x"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                </div>
              </div>
              <button
                class="link"
                href="/prefectures/27000"
              >
                大阪府
              </button>
            </li>
            <li
              class="item"
            >
              <div
                class="image"
              >
                <div
                  style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
                >
                  <div
                    style="box-sizing: border-box; display: block; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjM1IiBoZWlnaHQ9IjI1OSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                      style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                    />
                  </div>
                  <img
                    alt="兵庫県"
                    data-nimg="true"
                    decoding="async"
                    src="/_next/image?url=%2Fms-library%2Fimages%2Fthum_hyogo.jpg&w=640&q=75"
                    srcset="/_next/image?url=%2Fms-library%2Fimages%2Fthum_hyogo.jpg&w=256&q=75 1x, /_next/image?url=%2Fms-library%2Fimages%2Fthum_hyogo.jpg&w=640&q=75 2x"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                </div>
              </div>
              <button
                class="link"
                href="/prefectures/28000"
              >
                兵庫県
              </button>
            </li>
            <li
              class="item"
            >
              <div
                class="image"
              >
                <div
                  style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
                >
                  <div
                    style="box-sizing: border-box; display: block; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjM1IiBoZWlnaHQ9IjI1OSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                      style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                    />
                  </div>
                  <img
                    alt="愛知県"
                    data-nimg="true"
                    decoding="async"
                    src="/_next/image?url=%2Fms-library%2Fimages%2Fthum_aichi.jpg&w=640&q=75"
                    srcset="/_next/image?url=%2Fms-library%2Fimages%2Fthum_aichi.jpg&w=256&q=75 1x, /_next/image?url=%2Fms-library%2Fimages%2Fthum_aichi.jpg&w=640&q=75 2x"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                </div>
              </div>
              <button
                class="link"
                href="/prefectures/23000"
              >
                愛知県
              </button>
            </li>
            <li
              class="item"
            >
              <div
                class="image"
              >
                <div
                  style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
                >
                  <div
                    style="box-sizing: border-box; display: block; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjM1IiBoZWlnaHQ9IjI1OSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                      style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                    />
                  </div>
                  <img
                    alt="福岡県"
                    data-nimg="true"
                    decoding="async"
                    src="/_next/image?url=%2Fms-library%2Fimages%2Fthum_fukuoka.jpg&w=640&q=75"
                    srcset="/_next/image?url=%2Fms-library%2Fimages%2Fthum_fukuoka.jpg&w=256&q=75 1x, /_next/image?url=%2Fms-library%2Fimages%2Fthum_fukuoka.jpg&w=640&q=75 2x"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                </div>
              </div>
              <button
                class="link"
                href="/prefectures/40000"
              >
                福岡県
              </button>
            </li>
          </ul>
        </section>
        <section
          class="container"
        >
          <p
            class="title"
          >
            駅・路線から探す
          </p>
          <div
            class="main"
          >
            <button
              class="trigger"
            >
              駅・沿線から探す
            </button>
          </div>
        </section>
        <section
          class="container"
        >
          <p
            class="title"
          >
            マンション名から探す
          </p>
          <form
            action="/ms-library/freeword/"
            class="form"
            method="GET"
          >
            <input
              name="wd"
              placeholder="マンション名を入力してください"
              type="text"
              value=""
            />
            <button
              class="submit"
              type="submit"
            >
              検索
            </button>
          </form>
          <div
            class="tags"
          >
            <button
              class="link"
              href="/towns/38876"
            >
              小石川
            </button>
            <button
              class="link"
              href="/towns/38886"
            >
              本郷
            </button>
            <button
              class="link"
              href="/towns/38766"
            >
              高輪
            </button>
            <button
              class="link"
              href="/towns/38762"
            >
              白金
            </button>
            <button
              class="link"
              href="/towns/38775"
            >
              南麻布
            </button>
            <button
              class="link"
              href="/towns/38774"
            >
              南青山
            </button>
            <button
              class="link"
              href="/towns/39137"
            >
              駒沢
            </button>
            <button
              class="link"
              href="/towns/38759"
            >
              芝浦
            </button>
            <button
              class="link"
              href="/towns/38768"
            >
              西麻布
            </button>
            <button
              class="link"
              href="/towns/39044"
            >
              碑文谷
            </button>
            <button
              class="link"
              href="/towns/39187"
            >
              神宮前
            </button>
            <button
              class="link"
              href="/towns/38745"
            >
              晴海
            </button>
          </div>
        </section>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
